tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/FindReplaceDialogTest.java

```diff
@@ -426,6 +429,20 @@ public void testActivateWholeWordsAndSearchForTwoWords() {
 		assertFalse(dialog.wholeWordCheckBox.getEnabled());
 	}
 
+	@Test
+	public void testActivateDialogWithSelectionActive() {
+		openTextViewer("text" + System.lineSeparator() + "text" + System.lineSeparator() + "text");
+		fTextViewer.setSelectedRange(5, 9);
+		openFindReplaceDialogForTextViewer(false);
+
+		assertFalse(dialog.globalRadioButton.getSelection());
+		dialog.findCombo.setText("text");
+		select(dialog.replaceAllButton);
+		runEventQueue();
```

Comment:
Why should it be necessary to run the event queue here? The test fails on Linux despite this call. I think what is missing in general is to run the event queue once the action to open the find/replace dialog has been invoked (like is introduced in #1192).

Reply:
Root cause of the test failure will be fixed by #1836.

Reply:
Will #1836 really fix the test? From my understanding, the busy-indicator won't make this thread wait for performReplaceAll.

What we need for this test is a good way to know when the performReplaceAll-method is done running - no matter which thread it is in.

What is the correct procedure for right now?

Reply:
It will fix the test, because the test fails because of a `NullPointerException` when executing "replace all", as the active shell is still `null` (missing GTK event processing). Doing a `runEventQueue()` after initializing the dialog and before doing the "replace all" would also do the trick, but is rather a hack that will not be necessary after merging #1836.